src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/FeatureMapping.h

```diff
@@ -45,9 +46,17 @@ namespace OpenMS
       {
           public:
 
-          struct FeatureToMs2Indices
+          class FeatureMappingInfo
           {
-             Map<const BaseFeature*, std::vector<size_t>> assignedMS2;
+          public:
+            std::vector<FeatureMap> v_fp;
```

Comment:
please add documentation and some description. 
maybe rename the variables to something more telling.
std::vector<FeatureMap> feature_maps; // the actual feature data
KDTreeFeatureMaps kd_tree; // kdTree references into feature_maps and provides fast spatial queries